Replace recommonmark with myst (docs)

luizirber opened the pull request:
[MyST](https://myst-parser.readthedocs.io/en/latest/index.html) is a new markdown parser that supports features from ReStructureText, especially Sphinx directives and roles. This might be enough to replace the last `.rst` document (the index), for example, while also allowing more customization lost on the rst -> md transition.

## TODO

- [x] Convert `index.rst` to a markdown file
- ~[ ] Convert `api.rst` to a markdown file~ pending on https://github.com/executablebooks/MyST-Parser/issues/163

## Checklist

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

ctb commented:
question! does it flag missing local references for you, or are you just being extra careful in your review?

luizirber commented:
> question! does it flag missing local references for you, or are you just being extra careful in your review?

A bit of both. I replaced full URLs (pointing to `sourmash.readthedocs.org`), which allows finding internal references (and where they are linking to anchors that don't exist anymore). I also removed the recommendation to install pre-releases of sourmash 2.0 (!!!), and fixed some notebooks that had multiple `<h1>`-level headings (sphinx doesn't like that).

There are still many warnings about documents not properly linked in `toctrees`, I think it's useful to fix them and then use `sphinx-build -W` (turn warnings into errors) to avoid having any warnings pop up again.

luizirber commented:
heh, the myst parser doesn't support Python 2.7, so this broke Travis =P

luizirber commented:
Ready for review and merge @ctb @olgabot

ctb commented:
I'm not immediately sure why this is happening, but `make doc` no longer builds the docs on this branch, despite its presence in the Makefile.

ctb commented:
oh, I see the diff...  but that doesn't work for me :)

ctb commented:
I built it locally!

Questions:
- Do we want to add -W in this PR, or make an issue and do it in another PR? It looks like it works for me on a clean checkout.
- Has this been built successfully on RTD?

I haven't reviewed the content changes yet, will get into that next.

ctb commented on doc/developer.md:
```diff
@@ -205,3 +205,14 @@ For the Rust core library we use `rMAJOR.MINOR.PATH`
 (note it starts with `r`, and not `v`).
 The Rust version is not automated,
 and must be bumped in `src/core/Cargo.toml`.
+
```
Note that this TOC ends up at the bottom of the file. Is this intentional (and if so, why 😂)

luizirber replied:
I collected orphan links into a toc, but opted to put it in the bottom to avoid disrupting the current docs. But it might be more useful to put on the top...

ctb commented on doc/other-languages.md:
````diff
@@ -32,7 +32,8 @@ sourmash_comp_matrix <- as.matrix(sourmash_comp_matrix)
 ```
 
-[See the output of plotting and clustering this matrix](_static/ecoli-cmp.html), produced by [this RMarkdown file](_static/ecoli-cmp.Rmd).
+[See the output of plotting and clustering this matrix](https://sourmash.readthedocs.io/en/latest/_static/ecoli-cmp.html),
````
This is odd - we can't refer to local static HTML files?

luizirber replied:
Trying to find out, opened an issue in the MyST repo: https://github.com/executablebooks/MyST-Parser/issues/167

ctb replied:
punted to https://github.com/dib-lab/sourmash/issues/1051

ctb commented on doc/other-languages.md:
```diff
@@ -32,7 +32,8 @@ sourmash_comp_matrix <- as.matrix(sourmash_comp_matrix)
+produced by [this RMarkdown file](https://sourmash.readthedocs.io/en/latest/_static/ecoli-cmp.Rmd).
 
-You can download the `ecoli.cmp.csv` file itself [here](_static/ecoli.cmp.csv).
+You can download the `ecoli.cmp.csv` file itself [here](https://sourmash.readthedocs.io/en/latest/_static/ecoli.cmp.csv).
```
same, local static CSV files aren't reference-able? If not, we might consider pointing directly to the GitHub URL since they have a decent CSV renderer.

ctb replied:
punted to https://github.com/dib-lab/sourmash/issues/1051

ctb commented on doc/release-notes/releases.md:
```diff
@@ -0,0 +1,8 @@
+# sourmash release notes
```
Do we want to point to the github release notes in here? Dunno if we want to do it for each version, but maybe could add some text saying, "minor etc releases are documented <here>"

ctb replied:
punted to https://github.com/dib-lab/sourmash/issues/1052

ctb commented:
@olgabot re 

>Some places have docs plural and some have doc singular. For clarity, is it possible to have only one spelling?

I did some grepping in the markdown and can't find places where doc, singular, was used -- could you point us at a specific example? I'm on board with picking one just want to make sure I understand :)

ctb commented:
Issues that need to be resolved for a merge, IMO -

- [ ] make doc doesn't work
- [ ] how can we refer to local static HTML files? https://github.com/dib-lab/sourmash/pull/1021#discussion_r439840117
- [ ] how about local CSV files? https://github.com/dib-lab/sourmash/pull/1021#discussion_r439840177

I might work on these in a separate PR (into this one).

ctb commented:
`make doc` fixed in https://github.com/dib-lab/sourmash/pull/1053; the key problem showing up with `make -d` was,

```
  Successfully remade target file `build'.
 Finished prerequisites of target file `doc'.
 Prerequisite `build' is older than target `doc'.
```
so basically we just need to force the `cd doc && make html` command. One way to do that is to add .PHONY as a prereq.
